src/config.c

```diff
@@ -1059,25 +1049,18 @@ struct rewriteConfigState *rewriteConfigReadOldFile(char *path) {
 
         /* Not a comment, split into arguments. */
         argv = sdssplitargs(line,&argc);
-        if (argv == NULL) {
+        if (argv == NULL || !isConfigNameRegistered(argv[0])) {
```

Comment:
I'm not sure I love how this is being handled now, but we do no longer just remove configs from an old file.